README.md

```diff
@@ -25,16 +25,41 @@ Once you have implemented the `sa` program (and tested it to the best of your ab
 
 ## Algorithm
 
-*Which algorithm did you use to construct the suffix array. What is its running time?*
+In our project we have implemented the O(n²) algorithm that uses radix sort instead of the naive comparison based sorting.
+Bucket sort can be done in time O(n+k), where n is the length of our array (strings of length n have n or n+1 suffixes). The key is our alphabet, which for numbers would be digits from 0-9. In our case the alphabet is unknown but we have taken the liberty to assume that it is no more than a bytesize (256), since most normal characters fits in this range. This means k is constant and we can describe Bucket sort as O(n). The longest string we sort on has length n and therefore we need to do Bucket sort n times, resulting in the running time O(n²).
+
+For the search part we have implemented a binary search function that finds upper bounds and lower bounds. That variant of binary search runs O(m * log(n+z)) 
```

Comment:
It's impressive that you can output $z$ occurrences $O(\log z)$ time. Are you sure about that?